parser.go

```diff
@@ -94,14 +108,31 @@ func ParseMetric(packet []byte) (*Metric, error) {
 			if ret.Type == "gauge" || ret.Type == "set" {
 				return nil, fmt.Errorf("Invalid metric packet, %s cannot have a sample rate", ret.Type)
 			}
+			if sampleRate <= 0 {
+				return nil, fmt.Errorf("Sample rate %f must be positive", sampleRate)
```

Comment:
I’d prefer if we combined these so that the error message gives us a clear idea of the full requirements for `sampleRate`, e.g.,

```
if sampleRate <= 0 || sampleRate > 1 {
        return nil, fmt.Errorf("Sample rate %f should be > 0 and <= 1", sampleRate)
}
```